configure.bat

```diff
@@ -1,120 +1,180 @@
-@rem #    set to "path", then the executable will be the python.exe available
-@rem #    in the PATH.
+@rem # A configuration script to set things up: 
+@rem # create a virtualenv and install or update thirdparty packages.
+@rem # Source this script for initial configuration
+@rem # Use configure --help for details
+
+@rem # This script will search for a virtualenv.pyz app in etc\thirdparty\virtualenv.pyz
+@rem # Otherwise it will download the latest from the VIRTUALENV_PYZ_URL default
 @rem ################################
 
-@rem Current directory where this .bat files lives
-set CFG_ROOT_DIR=%~dp0
-@rem path where a configured Python should live in the current virtualenv if installed
-set CONFIGURED_PYTHON=%CFG_ROOT_DIR%tmp\Scripts\python.exe
-set PYTHON_EXECUTABLE=
 
+@rem ################################
+@rem # Defaults. Change these variables to customize this script
+@rem ################################
+
+@rem # Requirement arguments passed to pip and used by default or with --dev.
+set "REQUIREMENTS=--editable .[full]"
+set "DEV_REQUIREMENTS=--editable .[full,testing]"
+
+@rem # where we create a virtualenv
+set "VIRTUALENV_DIR=tmp"
+
+@rem # Cleanable files and directories to delete with the --clean option
+set "CLEANABLE=build tmp"
 
-@rem parse command line options and arguments
-:collectopts
-if "%1" EQU "--help" (goto cli_help)
-if "%1" EQU "--clean" (call rmdir /s /q "%CFG_ROOT_DIR%tmp") && call exit /b
-if "%1" EQU "--python" (set PROVIDED_PYTHON=%~2) && shift && shift && goto collectopts
+@rem # extra  arguments passed to pip
+set "PIP_EXTRA_ARGS= "
 
-@rem If we have a pre-configured Python in our virtualenv, reuse this as-is and run
-if exist ""%CONFIGURED_PYTHON%"" (
-    set PYTHON_EXECUTABLE=%CONFIGURED_PYTHON%
-    goto run
+@rem # the URL to download virtualenv.pyz if needed
+set VIRTUALENV_PYZ_URL=https://bootstrap.pypa.io/virtualenv.pyz
+@rem ################################
+
+
+@rem ################################
+@rem # Current directory where this script lives
+set CFG_ROOT_DIR=%~dp0
+set "CFG_BIN_DIR=%CFG_ROOT_DIR%\%VIRTUALENV_DIR%\Scripts"
+
+
+@rem ################################
+@rem # Set the quiet flag to empty if not defined 
+if not defined CFG_QUIET (
+    set "CFG_QUIET= "
 )
 
-@rem If we have a command arg for Python use this as-is
-if ""%PROVIDED_PYTHON%""==""path"" (
-    @rem use a bare python available in the PATH
-    set PYTHON_EXECUTABLE=python
-    goto run
+
+@rem ################################
+@rem # Main command line entry point
+set CFG_DEV_MODE=0
+set "CFG_REQUIREMENTS=%REQUIREMENTS%"
+
+if "%1" EQU "--help"   (goto cli_help)
+if "%1" EQU "--clean"  (goto clean)
+if "%1" EQU "--dev"    (
+    set "CFG_REQUIREMENTS=%DEV_REQUIREMENTS%"
+    set CFG_DEV_MODE=1
 )
-if exist ""%PROVIDED_PYTHON%"" (
-    set PYTHON_EXECUTABLE=%PROVIDED_PYTHON%
-    goto run
+if "%1" EQU "--python"  (
+    echo "The --python is now DEPRECATED. Use the PYTHON_EXECUTABLE environment

```

Comment:
there is an extra CR here.

Reply:
Thanks. Fixed in https://github.com/nexB/skeleton and merged here